### PR TITLE
ISPN-14622 ZSCAN

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14622

### CODE REVIEW

File: multimap/src/main/java/org/infinispan/multimap/impl/SortedSetBucket.java [hunk 1]
```diff
@@ -33,6 +34,10 @@ public class SortedSetBucket<V> {
    private final TreeSet<ScoredValue<V>> scoredEntries;
    private final Map<MultimapObjectWrapper<V>, Double> entries;
 
+   public Stream<ScoredValue<V>> stream() {
+      return scoredEntries.stream();
+   }
```
Comment (jabolina) [inline multimap/src/main/java/org/infinispan/multimap/impl/SortedSetBucket.java:39]: This seems unused.

File: server/core/src/main/java/org/infinispan/server/iteration/list/ListStreamSupplier.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      this.source = source;
+   }
+
+   @Override
+   public Stream<CacheEntry<Object, Object>> buildStream(IntSet ignore, Set valuesToFilter, boolean parallel) {
+      if (source == null) return Stream.empty();
+
+      Stream<Map.Entry<Object, Object>> stream;
+      if (valuesToFilter != null) {
+         stream = parallel ? valuesToFilter.parallelStream() : valuesToFilter.stream();
```
Comment (jabolina) [inline server/core/src/main/java/org/infinispan/server/iteration/list/ListStreamSupplier.java:34]: We need to instead filter the `source` stream here based on the `valuesToFilter`.

Comment (karesti) [reply server/core/src/main/java/org/infinispan/server/iteration/list/ListStreamSupplier.java:34]: yes but I was unable to run in that part in the tests!

Comment (karesti) [reply server/core/src/main/java/org/infinispan/server/iteration/list/ListStreamSupplier.java:34]: the keys to filter don't make sense in this one I think

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
CI failures are unrelated.

**Pull request merged**